test/integration/handlers/token-handler.spec.ts

```diff
@@ -0,0 +1,1670 @@
+import * as should from 'should';
+import * as util from 'util';
+import {
+  AccessDeniedError,
+  InvalidArgumentError,
+  InvalidClientError,
+  InvalidRequestError,
+  ServerError,
+  UnauthorizedClientError,
+  UnsupportedGrantTypeError,
+} from '../../../lib/errors';
+import { PasswordGrantType } from '../../../lib/grant-types';
```

Comment:
this test should cover
``
ImplicitGrantType, RefreshTokenGrantType, AuthorizationCodeGrantType, ClientCredentialsGrantType
```